api/handlers.go

```diff
@@ -31,3 +32,123 @@ func (api *API) userHandlerGET(w http.ResponseWriter, req *http.Request, _ httpr
 	}
 	WriteJSON(w, u)
 }
+
+// userUploadsHandler returns all uploads made by the current user.
+func (api *API) userUploadsHandler(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	sub, _, _, err := tokenFromContext(req)
+	if err != nil {
+		WriteError(w, err, http.StatusInternalServerError)
```

Comment:
Why are we returning 500 internal server errors here? In this particular case I would return `400 Bad Request`. I've always been thought it's bad API design to return 500 errors.

Reply:
I would go for "Unauthorized" for this particular case. It covers both the case where a token was not provided, did not contain the needed claims or was not valid.

We've talked about 500s before - I disagree that it's bad design to return those. An internal server error is a valid response that informs the user what happened. I do agree that we should strive to never return them but that's different from not having them on the API - if our service somehow failed then we should return a 500.

Again, in this case I agree that a 500 is not the best response, so I'm changing it to Unauthorized. If you think that bad Request is better then we can discuss that.